### ISSUE TITLE
git version 2.43.0 breaks git version checks

### ISSUE THREAD

**Issue by mattieruth** [newren/git-filter-repo#523]
In 2.43.0, the `git fast-export -h` output appears to have changed such that all the args are now prefixed with `[no-]` breaking the string parse checks [here](https://github.com/newren/git-filter-repo/blob/4bc9022afce5e2e138596bbecf4df310212ae164/git-filter-repo#L2076) and [here](https://github.com/newren/git-filter-repo/blob/4bc9022afce5e2e138596bbecf4df310212ae164/git-filter-repo#L2070). 

On 2.43.0:
```
$ git fast-export -h
...
--[no-]reencode <mode>
                          select handling of commit messages in an alternate encoding
--[no-]mark-tags      label tags with mark ids
```

On earlier version:
```
--reencode <mode>   select handling of commit messages in an alternate encoding
--mark-tags      label tags with mark ids
```

Note: `--anonymize-map` appears to remain unchanged and doesn't need fixing.

**Comment by jankap** [comment]
I can confirm that on Windows for git 2.43.0 :) Downgrading helped.

**Comment by kasperlewau** [comment]
Can confirm. 2.43.0 on Linux.

**Comment by TommiPrami** [comment]
How this affects on the usage of filter-repo? 

Does it work at all, or is this cosmetic problem?

I have 2.43.0 on windows and it seems to me that filter-repo does nothing, but does not give error message either.

**Comment by groboclown** [comment]
> How this affects on the usage of filter-repo?
> 
> Does it work at all, or is this cosmetic problem?
> 
> I have 2.43.0 on windows and it seems to me that filter-repo does nothing, but does not give error message either.

Certain filter-repo flags trigger checking if the flag is supported by your git version.  For example, if you run:

```
git filter-repo --path-rename "src/:target/" --preserve-commit-encoding
```

The logic behind checking for whether the flag is supported or not is done by looking at the output of `git fast-export -h`, which, in later versions of Git, use the text `--[no-]reencode`, whereas filter-repo looks for the text `--reencode`.

**Comment by newren** [comment]
Fixed two weeks ago with 7c5be8e30355 (Support git 2.43, 2023-12-08).  Sorry everyone for the long delay from my extended absence from the project.  :-(
